src/java/com/rapleaf/jack/BaseDatabaseConnection.java

```diff
@@ -8,20 +8,34 @@
 import java.sql.SQLException;
 import java.sql.Statement;
 
+import com.google.common.base.Optional;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public abstract class BaseDatabaseConnection implements Serializable, Closeable {
 
   private static Logger LOG = LoggerFactory.getLogger(BaseDatabaseConnection.class);
 
+  public static String DISABLE_PROPERTY = "jack.db.disallow_connections";
+  public static String DISABLE_ENV_VAR = "JACK_DB_DISALLOW_CONNECTIONS";
+
+
   protected transient Connection conn = null;
-  
+
   /**
    * Get a Connection to a database.
    * If there is no connection, create a new one.
    */
-  public abstract Connection getConnection();
+  public Connection getConnection() {
+    if (!isDisabled()) {
+      return getConnectionInternal();
+    } else {
+      throw new RuntimeException("Tried to instantiate a connection even though connection have been isDisabled in this environment");
```

Comment:
> have been isDisabled in this environment

hehe, intellij. few places below this in comments where this happened too. not that it really matters :)